src/length_delimited.rs

```diff
@@ -44,27 +42,38 @@ pub struct Builder {
 /// See [module level] documentation for more detail.
 ///
 /// [module level]: index.html
-pub struct Framed<T, B: IntoBuf = BytesMut> {
-    inner: FramedRead<FramedWrite<T, B>>,
-}
+pub type Framed<T> = codec::Framed<T, Codec>;
 
 /// Adapts a byte stream to a `Stream` yielding entire frame values.
 ///
 /// See [module level] documentation for more detail.
 ///
 /// [module level]: index.html
-#[derive(Debug)]
-pub struct FramedRead<T> {
-    inner: codec::FramedRead<T, Decoder>,
-}
+pub type FramedRead<T> = codec::FramedRead<T, Codec>;
+
+
+/// Adapts a byte stream to a `Sink` accepting entire frame values.
+///
+/// See [module level] documentation for more detail.
+///
+/// [module level]: index.html
+pub type FramedWrite<T> = codec::FramedWrite<T, Codec>;
 
 /// An error when the number of bytes read is more than max frame length.
 pub struct FrameTooBig {
     _priv: (),
 }
 
+/// A codec for frames delimited by a frame head specifying their lengths.
+///
+/// This allows the consumer to work with entire frames without having to worry
+/// about buffering or other framing logic.
+///
+/// See [module level] documentation for more detail.
+///
+/// [module level]: index.html
 #[derive(Debug)]
-struct Decoder {
+pub struct Codec {
```

Comment:
Since this will be the main type, maybe we should name it `LengthDelimited`. This way, when used, it will be `Framed<T, LengthDelimited>`.